apps/gdalalg_vector_change_field_type.cpp

```diff
@@ -0,0 +1,215 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  "change-field-type" step of "vector pipeline"
+ * Author:   Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "gdalalg_vector_change_field_type.h"
+#include "gdal_priv.h"
+#include "gdal_utils.h"
+
+//! @cond Doxygen_Suppress
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+/************************************************************************/
+/*          GDALVectorChangeFieldTypeAlgorithm::GDALVectorChangeFieldTypeAlgorithm()          */
+/************************************************************************/
+
+GDALVectorChangeFieldTypeAlgorithm::GDALVectorChangeFieldTypeAlgorithm(
+    bool standaloneStep)
+    : GDALVectorPipelineStepAlgorithm(NAME, DESCRIPTION, HELP_URL,
+                                      standaloneStep)
+{
+    auto &layerArg = AddActiveLayerArg(&m_activeLayer);
+    auto &fieldNameArg = AddFieldNameArg(&m_fieldName).SetRequired();
+    if (GetArg(GDAL_ARG_NAME_INPUT))
```

Comment:
the fact that we need to test this shows that in pipeline mode, completion will not work. We should probably pass instead the m_inputDataset member that should work both in standalone and pipeline mode

Reply:
Hmmm, I see that SetAutoCompleteFunctionForLayerName also receives a datasetArg:
```static void SetAutoCompleteFunctionForLayerName(
        GDALInConstructionAlgorithmArg &layerArg,
        GDALInConstructionAlgorithmArg &datasetArg);
```

Reply:
@rouault like this? https://github.com/OSGeo/gdal/pull/13086/commits/c90b1ef01b73bb55f8bb05bc6d324b0d7f95f64d#diff-07356a0dc0db470d267b20793311c46156d559ee08aeb20c908317945fec16e4R4499

Reply:
> > m_inputDataset has 0 items at that time, is it expected?
> 
> in pipeline or standalone mode ? 

In both modes, at construction time it is empty, but I guess that it is expected, when the autocomplete is actually run the dataset is filled with the input when testing `autotest/utilities/test_gdalalg_vector_change_field_type.py` so I guess we are good in standalone mode. 
I'm not sure how to test the pipeline mode.

Reply:
> I'm not sure how to test the pipeline mode.

```
gdal completion gdal pipeline read your.gpkg ! change-field-type --field-name
```

Reply:
m_inputDataset is empty in the above test.